src/mono/mono/metadata/icall.c

```diff
@@ -1802,8 +1802,8 @@ ves_icall_RuntimeTypeHandle_GetAttributes (MonoQCallTypeHandle type_handle)
 {
 	MonoType *type = type_handle.type;
 
-	if (m_type_is_byref (type) || type->type == MONO_TYPE_PTR || type->type == MONO_TYPE_FNPTR)
-		return TYPE_ATTRIBUTE_NOT_PUBLIC;
+	if (type->type == MONO_TYPE_FNPTR)
+		return TYPE_ATTRIBUTE_PUBLIC;
```

Comment:
It looks odd that we are unconditionally returning public for function pointers, but not for regular pointers.

Reply:
Yes it is odd, i guess the visibility should be same as the function it is pointing to, but at least for the test case i have added `Assert.True(typeof(delegate*<string, int>).IsPublic);` that is undefined, so i assumed it could be constant, though we might want to return the default, not public, and might need to add condition for the function it is pointing